lib/tasks/scripts/telemedicine_reports.rb

```diff
@@ -384,12 +384,14 @@ def facility_measures(facility, p_start, p_end)
     teleconsult_requests = facility
       .teleconsultations
       .where("device_created_at >= ? AND device_created_at <= ?", p_start, p_end)
-      .select("DISTINCT(patient_id)")
-    teleconsult_records = teleconsult_requests.where.not(recorded_at: nil)
-    teleconsult_marked_completed = teleconsult_requests.where(requester_completion_status: "yes")
-    teleconsult_marked_incomplete = teleconsult_requests.where(requester_completion_status: "no")
-    teleconsult_marked_waiting = teleconsult_requests.where(requester_completion_status: "waiting")
-    teleconsult_not_marked = teleconsult_requests.where(requester_completion_status: nil)
+      .select("DISTINCT ON (patient_id) patient_id, requester_completion_status, recorded_at")
+      .order(Arel.sql("patient_id, array_position(array['yes', 'no', 'waiting']::varchar[], requester_completion_status)"))
+      .to_a
+    teleconsult_records = teleconsult_requests.reject { |request| request.recorded_at.nil? }
+    teleconsult_marked_completed = teleconsult_requests.select { |request| request.requester_completion_status == "yes" }
+    teleconsult_marked_incomplete = teleconsult_requests.select { |request| request.requester_completion_status == "no" }
+    teleconsult_marked_waiting = teleconsult_requests.select { |request| request.requester_completion_status == "waiting" }
+    teleconsult_not_marked = teleconsult_requests.select { |request| request.requester_completion_status.nil? }
```

Comment:
Switching to in-memory counts because of the new distincting introduced above? This seems fine for now, but will it become a problem when our teleconsult activity grows larger?

Reply:
It could become problematic. This array is teleconsults distinct by patient, per facility, per week. Right now it tends to be < 10. If this starts to become a larger number that we continue tracking, we should probably move it over to a regular dashboard instead of this weekly csv export. We could use materialized views like we do with other reports to keep it scalable.

Reply:
Sounds good. Let's make sure that the activity (mem usage, code execution time, query performance) is showing up on DataDog already so we can reliably tell when this is no longer scalable :smile: